### DISCUSSION TITLE
Difference running on Windows vs Linux Docker Container

### DISCUSSION THREAD

**Discussion by brenosilva** [silverhack/monkey365#36] [Q&A]
Hello @silverhack,

I´m executing monkey365 both on a Docker Linux Container (host is an Ubuntu 22) and a Windows box.
Both using the same monkey365 configuration and i´m seeing a lot of "Unable to connect" messages on Docker side.

Please see the first 4 messages on both sides:

**Running from a Docker Linux Container:**
root@Ubuntu2204:/home/user/monkey365# docker run -it --env-file /tmp/myenvs.env monkey365 pwsh "/home/monkey365/monkey365/monkey365.ps1"
WARNING: Confidential application is not allowed to authenticate to Azure Portal AAD
**WARNING: Unable to connect To Exchange Online**
WARNING: Confidential application is not allowed to authenticate to https://ps.compliance.protection.outlook.com/Powershell-LiveId
**WARNING: Unable to connect to Exchange Compliance Center**
WARNING: [20:27:54:346] - [Test-EXOConnection] - Not connected to Exchange Online - warning - ee1c807af04c - ExoConnectionError

**Running from a Windows box**
WARNING: Confidential application is not allowed to authenticate to Azure Portal AAD
WARNING: Confidential application is not allowed to authenticate to https://ps.compliance.protection.outlook.com/Powershell-LiveId
WARNING: Your connection your redirected to URI: "https://nam12b.ps.compliance.protection.outlook.com/Powershell-LiveId?BasicAuthToOAuthConversion=true;PSVersion=5.1.22621.608"
WARNING:[20:32:57:723] - [Get-MonkeyEXOPhishFilterPolicy] - The Exchange Phish filter policy query did not return any data in ********************************* tenant - warning - user -
ExoPhishFilterResponse

The container can connect to internet hosts. 
Any idea what could be causing this connection errors on docker linux container?

**Comment by silverhack** [comment]
Hi @brenosilva hope you're doing well, 

It's not a Monkey365 issue. It's a [PowerShell issue](https://github.com/PowerShell/PowerShell/issues/7567). According to my tests, it only happens on Linux & MacOS and only with the Microsoft Purview service, so the recommendation would be to execute Monkey365 in Windows environments. 

Cheers,
